src/Shop.js

```diff
@@ -0,0 +1,84 @@
+class Shop {
+  constructor(items=[]){
+    this.items = items;
+  }
+
+  updateQuality() {
+    for (var i = 0; i < this.items.length; i++) {
+      var item = this.items[i];
+      var name = item.name;
+      if (name.includes('Sulfuras')) {
+        continue;
+      }
+      else if (name.includes('Aged Brie')) {
+        updateBrie(item);
+      }
+      else if (name.includes('Backstage passes')) {
+        updateConcertTicket(item);
+      } 
+      else if (name.includes('Conjured')) {
+        updateConjuredItem(item);
+      }
+      else {
+        updateRegularItem(item);
+      }
+      updateSellIn(item);
+      checkQualityBounds(item);
+    }
+    return this.items;
+
+    function updateSellIn(item) {
+      item.sellIn--;
+    }
+
+    function updateBrie(item) {
+      if (item.quality < 50) {
+        item.quality++;
+      }
+    }
+
+    function updateConcertTicket(item) {
+      if (item.sellIn <= 0) {
+        item.quality = 0;
+      }
+      else if (item.sellIn < 6) {
+        item.quality += 3
+      }
+      else if (item.sellIn < 11) {
+        item.quality += 2;
+      }
+      else {
+        item.quality++;
+      }
+    }
+    
+    function updateConjuredItem(item) {
+      if (item.sellIn <= 0) {
+        item.quality -= 4;
```

Comment:
Why are we degrading quality by 4? Was this part of the requirements?

Reply:
Regular items degrade by 2 after sellIn <=0. Conjured degrades twice as fast as Regular items.

Reply:
Cool, so lets have a test case for Conjured Items when the sellIn value is negative. It looks like you are only testing when it is positive.